fili-core/src/test/groovy/com/yahoo/bard/webservice/web/endpoints/JobsServletSpec.groovy

```diff
@@ -155,6 +192,35 @@ class JobsServletSpec extends Specification {
         GroovyTestUtils.compareJson(result, expectedResponse, JsonSortStrategy.SORT_BOTH)
     }
 
+    def "jobs/pagination/results returns the number of results we requested through pagination parameters" () {
+        setup:
+        String expectedResponse = """
+                                    {
+                                      "rows": [
+                                        {
+                                          "dateTime": "2016-01-12 00:00:00.000",
+                                          "pageViews": 111
+                                        }
+                                      ],
+                                      "meta": {
+                                        "pagination": {
+                                          "last": "http://localhost:9998/jobs/pagination/results?asyncAfter=5&perPage=1&page=3",
+                                          "next": "http://localhost:9998/jobs/pagination/results?asyncAfter=5&perPage=1&page=2",
+                                          "currentPage": 1,
+                                          "rowsPerPage": 1,
+                                          "numberOfResults": 3
+                                        }
+                                      }
+                                    }
+                                 """
+
+        when: "We send a request to the jobs/pagination/results endpoint"
+        String result = makeRequest("/jobs/pagination/results", [asyncAfter : ["5"], perPage : [1], page: [1] ])
+
+        then: "We get only first row as we requested against total number of rows"
+        GroovyTestUtils.compareJson(result, expectedResponse, JsonSortStrategy.SORT_BOTH)
```

Comment:
We're missing a few pagination cases:

1. We request a page in the middle.

2. We request a page at the end.

We should have cases for those as well.